src/test/java/de/tum/in/www1/artemis/architecture/ArchitectureTest.java

```diff
@@ -289,6 +292,30 @@ public void check(T item, ConditionEvents events) {
         };
     }
 
+    @Test
+    void testNoRequiredFalseOnMethodParams() {
+        methods().should(notHaveRequestParamWithRequiredFalse()).check(allClasses);
+    }
+
+    private ArchCondition<JavaMethod> notHaveRequestParamWithRequiredFalse() {
```

Comment:
This does not check if also a `defaultValue` parameter is set for the annotation. In such cases I would argue that defining the parameter as `required=false` together with a `defaultValue` is fine: Using an `Optional` would only introduce an unnecessary `Optional::get` call since we can be sure the parameter is always filled with a proper value.

Reply:
Yes you are correct, this should be fixed with [4f0aaf8](https://github.com/ls1intum/Artemis/pull/9146/commits/4f0aaf87dc2042c26aedf1fa9ce732b78d921456).

Reply:
The annotation class defines a default value when the user does not set one manually:

![image](https://github.com/user-attachments/assets/95944eb5-9d5c-4597-9e1c-016bdcf0e1a1)

Therefore, the new `annotation.get("defaultValue").isEmpty()` check always returns `true`.

Reply:
With [4f0aaf8](https://github.com/ls1intum/Artemis/pull/9146/commits/4f0aaf87dc2042c26aedf1fa9ce732b78d921456) I now compare against the default value of `defaultValue`. This also now produces less violations.

Reply:
Also thanks for the late-night reviews!